Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1263,6 +1317,77 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        /// <summary>
+        /// Gets or sets the threshold range that governs when the DataGrid class will begin to prefetch more items.
+        /// </summary>
+        /// <returns>
+        /// The loading threshold, in terms of pages.
+        /// </returns>
+        public double IncrementalLoadingThreshold
+        {
+            get { return (double)GetValue(IncrementalLoadingThresholdProperty); }
+            set { SetValue(IncrementalLoadingThresholdProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IncrementalLoadingThreshold"/> dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingThresholdProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingThreshold",
+                typeof(double),
+                typeof(DataGrid),
+                new PropertyMetadata(DATAGRID_defaultIncrementalLoadingThreshold, OnIncrementalLoadingThresholdPropertyChanged));
+
+        /// <summary>
+        /// IncrementalLoadingThresholdProperty property changed handler.
+        /// </summary>
+        /// <param name="d">DataGrid that changed its IncrementalLoadingThreshold.</param>
+        /// <param name="e">DependencyPropertyChangedEventArgs.</param>
+        private static void OnIncrementalLoadingThresholdPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            DataGrid dataGrid = d as DataGrid;
+            if (!dataGrid.IsHandlerSuspended(e.Property))
+            {
+                double oldValue = (double)e.OldValue;
+                double newValue = (double)e.NewValue;
+
+                if (double.IsNaN(newValue))
+                {
+                    dataGrid.SetValueNoCallback(e.Property, e.OldValue);
+                    throw DataGridError.DataGrid.ValueCannotBeSetToNAN(nameof(dataGrid.IncrementalLoadingThreshold));
+                }
+
+                if (newValue < 0)
+                {
+                    dataGrid.SetValueNoCallback(e.Property, e.OldValue);
+                    throw DataGridError.DataGrid.ValueMustBeGreaterThanOrEqualTo("value", nameof(dataGrid.IncrementalLoadingThreshold), 0);
+                }
+            }
```

Comment:
LoadMoreDataFromIncrementalItemsSource() should be called if the new value is larger than the old value.

Reply:
@RBrid Should be the same for OnDataFetchSizePropertyChanged as well right?

Reply:
@filipwa84, actually no. DataFetchSize is not a criteria that triggers an incremental load. It's only used after the decision to load more is made.